python/lsst/sims/catUtils/mixins/VariabilityMixin.py

```diff
@@ -640,7 +641,7 @@ def applyMLTflaring(self, valid_dexes, params, expmjd,
                                "knowledge of the effective area of the LSST "
                                "mirror.")
 
-        if _MLT_LC_NPZ is None or _MLT_LC_NPZ_NAME != self._mlt_lc_file:
+        if _MLT_LC_NPZ is None or _MLT_LC_NPZ_NAME != self._mlt_lc_file or _MLT_LC_NPZ.fid is None:
```

Comment:
I check for the fid attribute because, if someone runs sims_clean_up(), and then tries to simulate variability again, _MLT_LC_NPZ will not be `None`, but it will be closed (in which case, _MLT_LC_NPZ.fid will be `None`).